### PR TITLE
Add sqlflow.proto

### PR DESCRIPTION
**Pull request by wangkuiyi**

### CODE REVIEW

File: sqlflow.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  // represents all types of integer values in MySQL 8.0: TINYINT,
+  // INT, etc; FLOAT represent both fixed-point and floating-point and
+  // both 4-byte and 8-byte real values in MySQL 8.0.  DATE represents
+  // DATE, DATETIME, and TIMESTAMP.  STRING represents CHAR, VARCHAR,
+  // BINARY, VARBINARY, BLOB, TEXT.  More enum for other types like
+  // JSON and spatial in the future.
+  enum FieldType {
+    INT = 0; 
+    FLOAT = 1; 
+    DATE = 2;
```
Comment (ktong) [inline sqlflow.proto:46]: What’s format of date? ISO or Unix timestamp?

File: sqlflow.proto [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+message RunReply {
+  string error = 1;	 // An empty string indicates no error.
+  Table table  = 2;	 // `show databases` and `select` return data.
+}
+
+// Table represents the schema and rows of a RDBMA table.  It
+// represents the reply from SQL statements that return data.  Due to
+// the limit of protobuf message size
+// (https://stackoverflow.com/a/34186672/724872), it might represent
```
Comment (ktong) [inline sqlflow.proto:19]: We may use multiple data frames for exceed size if over http2

**Review by ktong — APPROVED** [review]
Have no idea how GRPC handle SSE. Need google it later

File: sqlflow.proto (hunk 2 above)
Comment (wangkuiyi) [reply sqlflow.proto:19]: I am afraid that we cannot. The size limit of protobuf is due to the encoding algorithm is not streaming, so there needs to be a buffer for encoding/decoding, and big messages might cause the buffer overflow.

File: sqlflow.proto (hunk 1 above)
Comment (wangkuiyi) [reply sqlflow.proto:46]: According to https://dev.mysql.com/doc/refman/8.0/en/datetime.html

> The DATE type is used for values with a date part but no time part. MySQL retrieves and displays DATE values in 'YYYY-MM-DD' format. The supported range is '1000-01-01' to '9999-12-31'.

> The DATETIME type is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in 'YYYY-MM-DD HH:MM:SS' format. The supported range is '1000-01-01 00:00:00' to '9999-12-31 23:59:59'.

> The TIMESTAMP data type is used for values that contain both date and time parts. TIMESTAMP has a range of '1970-01-01 00:00:01' UTC to '2038-01-19 03:14:07' UTC.

Comment (ktong) [reply sqlflow.proto:46]: So all of date’s timezone is UTC, right?

Comment (ktong) [reply sqlflow.proto:46]: Or client can translate to any timezone as date and date time has no timezone.

File: sqlflow.proto (hunk 2 above)
Comment (ktong) [reply sqlflow.proto:19]: gRPC streaming (https://grpc.io/docs/guides/concepts.html#server-streaming-rpc) definitely can handle big data.

Comment (ktong) [reply sqlflow.proto]: gRPC streaming (https://grpc.io/docs/guides/concepts.html#server-streaming-rpc) definitely can handle big data.

Comment (ktong) [reply sqlflow.proto:19]: The question here is if we want user download all data in one request. It consumes lot of server resource and kind of waste if client does not requests all of data. So most of api for retrieving data use pagination and only returns first page by default.

Comment (wangkuiyi) [reply sqlflow.proto:19]: 好。我按照 https://grpc.io/docs/tutorials/basic/go.html#server-side-streaming-rpc 试试。

**Pull request merged**